x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeListener.java

```diff
@@ -112,6 +112,7 @@ private ComputeListener(
             if (runningOnRemoteCluster()) {
                 // for remote executions - this ComputeResponse is created on the remote cluster/node and will be serialized and
                 // received by the acquireCompute method callback on the coordinating cluster
+                setFinalStatusAndShardCounts(clusterAlias, executionInfo);
```

Comment:
The two if-guards of `runningOnRemoteCluster()`  and `coordinatingClusterIsSearchedInCCS()` ensure that they are not null.  `clusterAlias` and `executionInfo` are both null when handling data node operations rather than coordinator level operations and CCS metadata is not tracked/updated during data node operations, only coordinator operations.